lib/pbench/test/unit/server/database/test_datasets_metadata_db.py

```diff
@@ -174,51 +175,55 @@ def test_server_keys(self, provide_metadata):
 
 
 class TestMetadataNamespace:
-    def test_get_bad_syntax(self, db_session, create_user):
-        ds = Dataset.create(owner=create_user.username, name="fio")
+    def test_get_bad_syntax(self, attach_dataset):
+        ds = Dataset.query(name="drb")
         with pytest.raises(MetadataBadKey) as exc:
             Metadata.getvalue(ds, "dashboard..foo")
         assert exc.type == MetadataBadKey
         assert exc.value.key == "dashboard..foo"
         assert str(exc.value) == "Metadata key 'dashboard..foo' is not supported"
 
-    def test_user_metadata(self, db_session, create_user, create_drb_user):
+    def test_user_metadata(self, attach_dataset):
         """Various tests on user-mapped Metadata keys"""
         # See if we can create a metadata row
-        ds = Dataset.create(owner=create_user.username, name="fio")
+        ds = Dataset.query(name="drb")
+        user1 = User.query(username="drb")
         assert ds.metadatas == []
-        t = Metadata.create(key="user", value=True, dataset=ds, user=create_user)
+        t = Metadata.create(key="user", value=True, dataset=ds, user=user1)
         assert t is not None
         assert ds.metadatas == [t]
-        assert create_user.dataset_metadata == [t]
+        assert user1.dataset_metadata == [t]
 
-        d = Metadata.create(key="user", value=False, dataset=ds, user=create_drb_user)
+        user2 = User.query(username="test")
+        d = Metadata.create(key="user", value=False, dataset=ds, user=user2)
         assert d is not None
         assert ds.metadatas == [t, d]
-        assert create_user.dataset_metadata == [t]
-        assert create_drb_user.dataset_metadata == [d]
+        assert user1.dataset_metadata == [t]
+        assert user2.dataset_metadata == [d]
 
         g = Metadata.create(key="user", value="text", dataset=ds)
         assert g is not None
         assert ds.metadatas == [t, d, g]
-        assert create_user.dataset_metadata == [t]
-        assert create_drb_user.dataset_metadata == [d]
+        assert user1.dataset_metadata == [t]
+        assert user2.dataset_metadata == [d]
 
         assert Metadata.get(key="user", dataset=ds).value == "text"
-        assert Metadata.get(key="user", dataset=ds, user=create_user).value is True
-        assert Metadata.get(key="user", dataset=ds, user=create_drb_user).value is False
+        assert Metadata.get(key="user", dataset=ds, user=user1).value is True
+        assert Metadata.get(key="user", dataset=ds, user=user2).value is False
 
-        Metadata.remove(key="user", dataset=ds, user=create_drb_user)
-        assert create_drb_user.dataset_metadata == []
-        assert create_user.dataset_metadata == [t]
-        assert ds.metadatas == [t, g]
+        Metadata.remove(key="user", dataset=ds, user=user1)
+        assert user1.dataset_metadata == []
+        assert user2.dataset_metadata == [d]
+        assert ds.metadatas == [d, g]
```

Comment:
Making the change to use the standard fixture datasets seemed like a good idea because I didn't have to add resource IDs in a lot of places; but it also led to some messy unraveling as the references changed here. Perhaps I could have done it differently. Perhaps that would have removed a few diffs. I didn't think it was worth any additional effort. I still don't.